app.js

```diff
@@ -8,17 +8,38 @@ var mongoose = require('mongoose');
 var routes = require('./routes');
 var logger = require('./config/logger');
 var errors = require('@feathersjs/errors');
+import jwt from 'jsonwebtoken';
+import passport from 'passport';
+import { ExtractJwt, Strategy as JwtStrategy } from 'passport-jwt';
+import User from './models/users.model';
 
 var app = express();
 
-// uncomment after placing your favicon in /public
-//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
 app.use(morgan('dev'));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
 
-app.use('/api', routes);
+// setting up passport
+var jwtOptions = {}
+jwtOptions.jwtFromRequest = ExtractJwt.fromAuthHeaderAsBearerToken();
+jwtOptions.secretOrKey = 'MyS3cr3t';
+
+var strategy = new JwtStrategy(jwtOptions, function(jwt_payload, next) {
+  logger.debug('payload received', jwt_payload);
+  User.read(jwt_payload.userId)
+    .then(user => {
+      if (user) {
+        return next(null, user);
+      } else {
+        return next(null, false);
+      }
+    });
+});
+
+passport.use(strategy);
+
+app.use('/api', passport.authenticate('jwt', { session: false }), routes);
```

Comment:
Em chưa hiểu chỗ này đại ca ơi, authenticate jwt ở đây thì làm sao vào được login nhỉ, sau khi login xong mới có jwt mà

Reply:
A đang sửa lại cái này đây, hôm trước xem lại thấy thốn quá 😢. Anh sẽ làm authenticate cho từng route, nhưng thế duplicate nhiều chỗ quá, route nào cần đều phải thêm `passport.authenticate('jwt', { session: false })`. Có suggest gì không e ?

Reply:
cái này em cũng chưa có giải pháp gì, chỉ thấy thế nên hỏi thôi :))

Reply:
Thế e cứ để authenticate cho từng route vậy, stupid but it works =))